### PR TITLE
Merge the core servers

### LINKED ISSUE #43: Merge the core services
Unfortunately, splitting the auth- and resource-server didn't really work out. Especially in view of the migration to a standalone application (#5), it makes sense to just have 1 artifact. Other reasons are:
- The communication between auth- and resource-server increases the overall complexity and introduces a bottleneck within the system
- Both services are tightly coupled by design, leaving a split questionable
- The core OSIAM services should feel like "one thing" not a distributed system in itself
- Versioning is sometimes confusing
- Building, testing and deploying is more complex than necessary
- Documentation, issues and other information is scattered across multiple repos
- Migration to BCrypt hashes becomes possible

This list makes no claims of being complete.

I'd also suggest, that we merge the functionality of addon-self-administration and addon-administration into the core, too. But I would leave that for the future.

As splitting once had a use case, namely operating the user-facing auth-server part on a public interface, but hide the resource-server part in a private network, we should think about
1. whether we still want to support this use case
2. how we want to implement it

To answer 1.: I don't think so. After all, controlling access is OSIAM's business, so there should be no harm in having the resource-server's endpoint publicly available. They're secured with OAuth anyway. I'm also not aware of that anyone is operating OSIAM like this. One can still simulate this "feature" by putting OSIAM in a private network and reverse proxying just the auth-server endpoints, or something like that.

We should put this on the 3.0 Roadmap, as it is a major breaking change.

### PR DESCRIPTION
**Pull request by tkrille**
For the rationale behind this see #43.

Open tasks:
- [x] Get ITs working
- [x] Adjust unit tests
- [x] Migrate commits https://github.com/osiam/resource-server/commit/47f2cd818321528a04c4505f67f8c14368f66898, https://github.com/osiam/resource-server/commit/8707d809d8ba7b92b30d3b86432fc6558f67fba5
- [x] Rename assets' base dir (is `auth-server/`) has become `osiam/`
- [x] Merge and adjust docs
- [x] Adjust `README.md`
- [x] Create `CHANGELOG.md`
- [x] Find solution for `auth-server` extension. See https://github.com/osiam/osiam/issues/47
## Changes in other repos
- [x] connector4java - https://github.com/osiam/connector4java/pull/176
- [x] addon-administration - https://github.com/osiam/addon-administration/pull/110
- [x] addon-self-administration - https://github.com/osiam/addon-self-administration/pull/156
- [x] shell - https://github.com/osiam/shell/pull/21
- [x] examples - https://github.com/osiam/examples/pull/24
- [x] ITs - https://github.com/osiam/connector4java-integration-tests/pull/303
- [x] performance tests osiam/performance-tests#5
- [x] Docker image - https://github.com/osiam/docker-image/pull/26
- [x] Vagrant - https://github.com/osiam/vagrant/pull/20
- [x] distribution - https://github.com/osiam/distribution/pull/37

Resolves #43

### CODE REVIEW

**Comment by tkrille** [comment]
First changes in other repos arrived: connector, self-admin, ITs. Also, the unit tests are running now.

So, now you can test this change.

**Comment by tkrille** [comment]
Just finished merging the docs.

**Comment by dacrome** [comment]
Updated performance-tests task and link PR

**Comment by tkrille** [comment]
I don't know why the ITs are only running on my machine. I even removed everything related from my machine and cloned a fresh copy of the 4 repos, but still the tests are successful. What else can we try to find the cause of this really strange problem?

**Comment by tkrille** [comment]
I think I solved the mysterious IT issue: We had 2 exception handlers that both handled the generic `Exception.class`: `org.osiam.auth.exception.WebExceptionHandler` and `org.osiam.resources.exception.RestExceptionHandler`. Oddly enough, it seems, that on my machine the `RestExceptionHandler` is the one, who's asked first in case of an unhandled exception, and on other machines it's `WebExceptionHandler`. What I did to solve it, was to restrict both exception handlers to the controllers they should actually watch over. Please re-test.

**Comment by tkrille** [comment]
Migrated the new commits from resource-server.

**Comment by tkrille** [comment]
What about the auth-server extension? The URN reads `urn:org.osiam:scim:extensions:auth-server`. Should be changed to `urn:org.osiam:scim:extensions:osiam` or `urn:org.osiam:scim:extensions:auth` or something? Seems safe to automatically migrate with Flyway.

**Comment by tkrille** [comment]
Updated Docker image repo.

**Comment by tkrille** [comment]
Regarding the auth-server extension: I will create a separate issue for it.

**Comment by tkrille** [comment]
This is final now, AFAIC.

**Comment by tkrille** [comment]
After your review @dacrome I will squash the top commits. Please give me a notice as soon you're ready.

**Comment by tkrille** [comment]
As a hint: after the merge we must activate this repo on CircleCI and create a Jenkins job on the internal CI.

**Comment by dacrome** [comment]
Updated Shell PR

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+    <name>OSIAM</name>
+    <description>
+        OSIAM is a secure identity management solution providing REST based services for authentication and authorization. We achieve this by implementing two important open standards:
+
+        * OAuth 2.0
+        * SCIM v2 (still in draft phase)
+
+        OSIAM is published under the MIT licence, giving you the greatest freedom possible to utilize OSIAM in you project or product.
+    </description>
+    <url>https://github.com/osiam/auth-server</url>
```
Comment (dacrome) [inline pom.xml]: https://github.com/osiam/osiam

**Comment by dacrome** [comment]
After you changed the one line I commented, please squash it and I will merge it! Yeah!

**Comment by tkrille** [comment]
amended and squashed

**Comment by dacrome** [comment]
> As a hint: after the merge we must activate this repo on CircleCI and create a Jenkins job on

Already activated on circleci. What about the jenkins job?

**Comment by tkrille** [comment]
Jenkins: After this has been merged.

**Comment by tkrille** [comment]
I would recommend that you re-test this, because of the squash and so on.

**Comment by dacrome** [comment]
Re-tested. All tests are green! Thanks @tkrille for your awesome work!

**Pull request merged**